packages/amplify-provider-awscloudformation/src/__tests__/utils/admin-login-server.test.ts

```diff
@@ -0,0 +1,44 @@
+import { AdminLoginServer } from '../../utils/admin-login-server';
+import { $TSContext } from 'amplify-cli-core';
+import { FunctionRuntimeLifecycleManager } from 'amplify-function-plugin-interface';
+
+const runtimePlugin_stub = ({
+  checkDependencies: jest.fn().mockResolvedValue({ hasRequiredDependencies: true }),
+  build: jest.fn().mockResolvedValue({ rebuilt: true }),
+} as unknown) as jest.Mocked<FunctionRuntimeLifecycleManager>;
+
+const context_stub = ({
+  amplify: {
+    readBreadcrumbs: jest.fn().mockReturnValue({ pluginId: 'testPluginId' }),
+    loadRuntimePlugin: jest.fn().mockResolvedValue(runtimePlugin_stub),
+    updateamplifyMetaAfterBuild: jest.fn(),
+  },
+} as unknown) as jest.Mocked<$TSContext>;
```

Comment:
Only `context.print` method is passed to the `AdminLoginServer` constructor, there's no need to mock the extra parts of the `context` object.

Reply:
Good point. We could make context smaller! Let me make follow up PR shortly to make the test simple. Thank you!